### PR TITLE
Meta: Script to configure clangd according to build type

### PR DESCRIPTION
**Pull request by hawkinsw**
Add a simple shell script to update the local clangd configuration according to the type of build selected by the user.

### CODE REVIEW

**Comment by hawkinsw** [comment]
This is my first attempt at a contribution and I have attempted to follow all the instructions! This little script was something that I needed and thought that others might need it, too! If people think it would be helpful, that's great! I am more than happy to make any changes necessary.

I am excited to attempt my first contribution. Thank you _all_ for working on such an awesome project.

File: Meta/configure-clangd.sh [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        build_type="-$2"
+        ;;
+    release)
+        build_type=""
+        ;;
+    *)
+        echo "Invalid build configuration specified: $2"
+        usage
+        exit 1
+esac
```
Comment (ADKaster) [inline Meta/configure-clangd.sh:45]: Hm, these don't quite align with the build types and directories set in CMakePresets.json.

Those being:

default (RelWithDebInfo) : Build/ladybird
Debug (Debug): Build/ladybird-debug
Sanitizer: Build/ladybird-sanitizers

That being said, we are missing a `build` and `test` preset for the `Debug` configure preset.

I will note that in this project basically none of the regular contributors are familiar with or use the multi-config setups common in Windows or Xcode centric environments, or Ninja multi-config.

Is it common to want to switch your clangd configuration to match either release or debug? Personally I just always have a `default` build up to date, and build debug or sanitizers as needed 🤔

**Review by ADKaster — COMMENTED** [review]
Welcome! This script seems interesting, but I'm not sure if I would use it myself.. 

If it is useful to you though, some documentation (perhaps in AdvancedBuildInstructions, or the documentation for your favorite editor?) would probably help other contributors understand when it would be useful to them.

**Comment by hawkinsw** [comment]
> Welcome! This script seems interesting, but I'm not sure if I would use it myself..
> 
> If it is useful to you though, some documentation (perhaps in AdvancedBuildInstructions, or the documentation for your favorite editor?) would probably help other contributors understand when it would be useful to them.

Thank you for the feedback! I will absolutely add this information to that documentation (and I will respond to your other helpful feedback to the PR). Thank you, again, for the feedback!

**Comment by hawkinsw** [comment]
@ADKaster I hope that my changes addressed your comments. Thanks again for all that you are doing to help with Ladybird!

**Comment by hawkinsw** [comment]
@ADKaster I think that I got CI happy. Sorry for the delay!

File: Documentation/AdvancedBuildInstructions.md [hunk 2]
```diff
@@ -67,3 +67,7 @@ Some OS distributions don't ship bleeding-edge clang-format binaries. Below are
 
 1) If you have a Debian-based (apt-based) distribution, use the [LLVM apt repositories](https://apt.llvm.org) to install the latest release of clang-format.
 2) Compile LLVM from source as described in the LLVM documentation [here](https://llvm.org/docs/GettingStarted.html#compiling-the-llvm-suite-source-code).
+
+## Clangd Configuration
+
+Clangd will automatically look for configuration information in files named `.clangd` in each of the parent directories of the file being edited. The Ladybird source code repository has a top-level `.clangd` configuration file in the root directory. One of the configuration stanzas in that file specifies the location fo a compilation database. Depending on your build configuration (e.g., Debug, default, Sanitizer, etc), the path to the compilation database in that file may not be correct. The result is that clangd will have a difficult time understanding all your include directories. To resolve the problem, you can use the `Meta/configure-clangd.sh` script.
```
Comment (ADKaster) [inline Documentation/AdvancedBuildInstructions.md:73]: typo: `the location fo a compilation database`

While you're here, could you wrap this at ~100 to 120 columns or so? It has no impact on the rendered markdown but is a bit easier on the eyes in an editor without line wrapping turned on, and in the github review area.

**Review by ADKaster — COMMENTED** [review]
The explanation makes sense, thanks! Hopefully this helps other folks that run into the problem you were avoid any confusion.

File: Documentation/AdvancedBuildInstructions.md (hunk 2 above)
Comment (hawkinsw) [reply Documentation/AdvancedBuildInstructions.md]: Sorry about the typo! I have fixed the typo and adjusted the layout!

**Comment by hawkinsw** [comment]
Sorry again about the typo! I am hoping this contribution is my onramp to additional contributions!

**Pull request merged**